refextract/app.py

```diff
@@ -0,0 +1,101 @@
+from flask import Flask, jsonify, make_response
+from webargs import fields
+from webargs.flaskparser import FlaskParser
+
+from refextract.references.api import (extract_journal_reference,
+                                       extract_references_from_file,
+                                       extract_references_from_string)
+from refextract.utils import download_file_from_url
+
+parser = FlaskParser()
+
+
+def create_app():
+    app = Flask(__name__)
+    app.config.from_pyfile("condig.cfg", silent=True)
+
+    @app.route("/extract_journal_info", methods=["POST"])
+    @parser.use_args(
+        {
+            "publication_infos": fields.List(fields.Dict, required=True),
+            "journal_kb_data": fields.Dict(required=True),
```

Comment:
it seems a bit silly to me to send a ~1MB journal data payload on each request (and do the serialization-deserialization several times between redis->hep->refextract), especially because refextract has its own caching mechanism in https://github.com/inspirehep/refextract/blob/master/refextract/references/kbs.py#L67-L71. Have you thought about other mechanisms, like exposing the journal kb on `hep` through an HTTP endpoint with proper ETag handling, pass the URL instead here and have refextract request the kb with an `If-None-Match` header set to the last ETag it has seen? I know it's more work and it might not be needed, but this feels a bit dirty.

Reply:
We have thought different options, at the moment we don't want to overcomplicate the transition to this service. Hence we will simply pass the kb over the network.